scripts/rv_expression_association/get_vep_variants.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import logging
+import hail as hl
+import numpy as np
+import pandas as pd
+from hail.methods import export_plink
+from cpg_utils.hail_batch import dataset_path, init_batch, output_path
+
+# object containing variants within a 50K window on either side of the VPREB3 gene
+MT = dataset_path('v0/VPREB3_50K_window.mt')
+VEP_HT = dataset_path('v0/VPREB3_50K_window_vep.ht')
+
+# use logging to print statements, display at info level
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+
+
+def main():
+    # read and densify object
+    init_batch()
+    mt = hl.read_matrix_table(MT)
+    mt = hl.experimental.densify(mt)
+
+    logging.info('Number of variants in window: {}'.format(mt.count()[0]))
+
+    # filter out low quality variants
+    mt = mt.filter_rows(hl.len(hl.or_else(mt.filters, hl.empty_set(hl.tstr))) == 0)
+
+    # consider biallelic variants only (no multi-allelic, no ref-only)
+    mt = mt.filter_rows(hl.len(mt.alleles) == 2)
+    mt = mt.filter_rows(hl.is_snp(mt.alleles[0], mt.alleles[1]))
+
+    # annotate using VEP
+    vep_ht = hl.read_table(VEP_HT)
+    mt = mt.annotate_rows(vep=vep_ht[mt.row_key].vep)
+
+    logging.info('Number of QC-passing, biallelic SNPs: {}'.format(mt.count()[0]))
+
+    # filter rare variants only (MAF < 5%)
+    mt = hl.variant_qc(mt)
+    rv_mt = mt.filter_rows(mt.variant_qc.AF[1] < 0.05)
+    rv_mt = rv_mt.filter_rows(rv_mt.variant_qc.AF[1] > 0)
+
+    logging.info('Number of rare variants (freq<5%): {}'.format(rv_mt.count()[0]))
```

Comment:
Again, the `count` operation will lead to all the operations being run. Consider using `checkpoint` to avoid that, if you need the count here (and in the log statement below).

Reply:
Maybe I could save these stats (instead of printing them out), if the count() operation is especialy expensive?

Reply:
`count` itself isn't particularly expensive -- what's expensive is that it triggers the evaluation of the tree of operations that are lazily executed in Hail. The results of that are _not_ automatically cached. E.g. consider this example:

```py
mt = hl.read_matrix_table(...)
mt = mt.filter_rows(...)
print(mt.count())
print(mt.count())
```

This will run `filter_rows` _twice_, which might be counterintuitive at first, but is a result of the lazy evaluation of expressions: https://hail.is/docs/0.2/overview/expressions.html

Instead, you can rewrite this as follows:

```py
mt = hl.read_matrix_table(...)
mt = mt.filter_rows(...)
mt = mt.checkpoint(...)
print(mt.count())
print(mt.count())
```

This explicitly writes out the intermediate result (which will use storage, but the `tmp` bucket is a good destination for this), avoiding the repeated evaluation of `filter_rows` in order to evaluate `count`.

Reply:
ok, i've tried to add checkpoints, maybe check if i missed any?

Reply:
You got almost all of them :). Flagged two more spots.

Reply:
should be all done now? The other comment you have is outdated I think, I did delete that print statement in my last push